packages/core/src/shared/regions/regionProvider.ts

```diff
@@ -162,13 +162,20 @@ export class RegionProvider {
         remote: () => Endpoints | Promise<Endpoints>
     }): RegionProvider {
         const instance = new this()
+        void instance.init(endpointsProvider)
+        return instance
+    }
 
-        async function load() {
+    async init(endpointsProvider: {
```

Comment:
Should we switch this to a try/catch at this point? Dropping the `const load`

Reply:
just played around with this. let me know what you think